infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -276,8 +284,19 @@ public synchronized void handleMessageLayoutPropose(
                     LayoutProposeResponse(phase1Rank.getRank())));
             return;
         }
-        Rank phase2Rank = getPhase2Rank();
-        Layout proposeLayout = msg.getPayload().getLayout();
+
+        final Rank phase2Rank = getPhase2Rank(payloadEpoch);
+        final Layout proposeLayout = msg.getPayload().getLayout();
+
+        // Make sure that the layout epoch is the same as the LayoutProposeRequest epoch.
+        if (proposeLayout.getEpoch() != payloadEpoch) {
+            log.debug("Phase II error: layout {} and payload {} epoch should be the same",
```

Comment:
Prepend log message with `handleMessageLayoutPropose` to keep it consistent with the rest of the method.

Reply:
I will be changing it in the next patch. There are a lot of formatting issues in this file.